File: app/src/main/java/com/twiceyuan/galleryviewpager/GalleryTransformer.java
package com.twiceyuan.galleryviewpager;

import android.support.v4.view.ViewPager;
import android.view.View;

public class GalleryTransformer implements ViewPager.PageTransformer {
    @Override
    public void transformPage(View view, float position) {
        float scale = 0.5f;
        float scaleValue = 1 - Math.abs(position) * scale;
        view.setScaleX(scaleValue);
        view.setScaleY(scaleValue);
        view.setAlpha(scaleValue);
        view.setPivotX(view.getWidth() * (1 - position - (position > 0 ? 1 : -1) * 0.75f) * scale);
        view.setElevation(position > -0.25 && position < 0.25 ? 1 : 0);
    }
}

File: app/src/main/java/com/twiceyuan/galleryviewpager/GalleryTransformer.kt
package com.twiceyuan.galleryviewpager

import android.support.v4.view.ViewPager
import android.view.View

object GalleryTransformer : ViewPager.PageTransformer {

    private const val scale = 0.5f

    override fun transformPage(view: View, position: Float) {
        val scaleValue = 1 - Math.abs(position) * scale
        view.scaleX = scaleValue
        view.scaleY = scaleValue
        view.alpha = scaleValue
        view.pivotX = view.width * (1f - position - if (position > 0) 0.75f else -0.75f) * scale
        view.pivotY = view.height / 2f
        view.elevation = 1f - Math.abs(position)
    }
}
